File: source/clibCore/addonCollection_adminSite/openAjaxIndexFilterGetContent.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.AdminSite
    Public Class openAjaxIndexFilterGetContentClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core

                '
                Call cpCore.visitProperty.setProperty("IndexFilterOpen", "1")
                Dim adminSite As New Contensive.Addons.AdminSite.getAdminSiteClass(cpCore.cp_forAddonExecutionOnly)
                Dim ContentID As Integer = cpCore.docProperties.getInteger("cid")
                If ContentID = 0 Then
                    result = "No filter is available"
                Else
                    Dim cdef As Models.Complex.cdefModel = Models.Complex.cdefModel.getCdef(cpCore, ContentID)
                    result = adminSite.GetForm_IndexFilterContent(cdef)
                End If
                adminSite = Nothing


            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: source/processor/Addons/AdminSite/bak/openAjaxIndexFilterGetContent.cs


using Contensive.BaseClasses;
using Contensive.Core;
using Models.Entity;
using Controllers;

// 

namespace Contensive.Addons.AdminSite {
    
    public class openAjaxIndexFilterGetContentClass : Contensive.BaseClasses.AddonBaseClass {
        
        // 
        // ====================================================================================================
        // '' <summary>
        // '' getFieldEditorPreference remote method
        // '' </summary>
        // '' <param name="cp"></param>
        // '' <returns></returns>
        public override object execute(Contensive.BaseClasses.CPBaseClass cp) {
            string result = "";
            try {
                CPClass processor;
                cp;
                CPClass;
                coreClass cpCore = processor.core;
                // 
                cpCore.visitProperty.setProperty("IndexFilterOpen", "1");
                Contensive.Addons.AdminSite.getAdminSiteClass adminSite = new Contensive.Addons.AdminSite.getAdminSiteClass(cpCore.cp_forAddonExecutionOnly);
                int ContentID = cpCore.docProperties.getInteger("cid");
                if ((ContentID == 0)) {
                    result = "No filter is available";
                }
                else {
                    Models.Complex.cdefModel cdef = Models.Complex.cdefModel.getCdef(cpCore, ContentID);
                    result = adminSite.GetForm_IndexFilterContent(cdef);
                }
                
                adminSite = null;
            }
            catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            
            return result;
        }
    }
}